feature-libs/asm/customer-360/components/sections/asm-customer-customer-coupon/asm-customer-customer-coupon.component.ts

```diff
@@ -0,0 +1,173 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import {
+  Customer360CustomerCouponList,
+  Customer360Facade,
+  Customer360Type,
+} from '@spartacus/asm/customer-360/root';
+import { CustomerCouponService } from '@spartacus/core';
+import {
+  BehaviorSubject,
+  Observable,
+  Subscription,
+  combineLatest,
+  of,
+} from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { Customer360SectionContext } from '../customer-360-section-context.model';
+import { CustomerCouponEntry } from './asm-customer-customer-coupon.model';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'cx-asm-customer-coupon',
+  templateUrl: './asm-customer-customer-coupon.component.html',
+})
+export class AsmCustomerCustomerCouponComponent implements OnInit, OnDestroy {
+  showErrorAlert$ = new BehaviorSubject<boolean>(false);
+  showErrorAlertForApplyAction$ = new BehaviorSubject<boolean>(false);
+  entries$: Observable<Array<CustomerCouponEntry>>;
+  subscription = new Subscription();
+  currentTabIsAssignable = true;
+
+  constructor(
+    protected context: Customer360SectionContext<Customer360CustomerCouponList>,
+    protected customer360Facade: Customer360Facade,
+    protected customerCouponService: CustomerCouponService
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.customerCouponService
+        .getClaimCustomerCouponResultError()
+        .subscribe((error) => {
+          if (error) {
+            this.changeTab(undefined, undefined);
+            this.showErrorAlertForApplyAction$.next(true);
+          }
+        })
+    );
+    this.subscription.add(
+      this.customerCouponService
+        .getDisclaimCustomerCouponResultError()
+        .subscribe((error) => {
+          if (error) {
+            this.changeTab(undefined, undefined);
+            this.showErrorAlertForApplyAction$.next(true);
+          }
+        })
+    );
+    this.fetchCustomerCoupons();
+    this.currentTabIsAssignable = true;
+    this.showErrorAlert$.next(false);
+    this.showErrorAlertForApplyAction$.next(false);
+  }
+
+  public fetchCustomerCoupons(): void {
+    this.entries$ = combineLatest([this.context.data$]).pipe(
+      map(([data]) => {
+        const entries: Array<CustomerCouponEntry> = [];
+        data.customerCoupons.forEach((customerCoupon) => {
+          entries.push({
+            code: customerCoupon.name,
+            name: customerCoupon.description,
+            codeForApplyAction: customerCoupon.code,
+            applied: false,
+          });
+        });
+        return entries;
+      }),
+      catchError(() => {
+        this.showErrorAlert$.next(true);
+        return of([]);
+      })
+    );
+  }
+
+  public changeTab(
+    assignable: boolean | undefined,
+    searchQuery: string | undefined
+  ): void {
```

Comment:
- Sometimes, user just want to use search without change tab, then `assignable` is undefined. In this scenario, `currentTabIsAssignable` is used for `assignable`, see line 103.
- When user change tab, `assignable` has value, then `assignable` is used. see line 101.